buildscripts/build-native.cmd

```diff
@@ -46,15 +53,24 @@ pushd "%__IntermediatesDir%"
 call "%__SourceDir%\Native\gen-buildsys-win.bat" "%__ProjectDir%\src\Native" %__VSVersion% %__BuildArch% 
 popd
 
-if exist "%__IntermediatesDir%\install.vcxproj" goto BuildNative
+if exist "%__IntermediatesDir%\install.vcxproj" goto BuildNativeVs
 echo Failed to generate native component build project!
 exit /b 1
 
-:BuildNative
+:PrepareEmscripten
+:: TODO: Add real wasm preparation
+goto :BuildNativeEmscripten
+
+:BuildNativeVs
 %_msbuildexe% /ConsoleLoggerParameters:ForceNoAlign "%__IntermediatesDir%\install.vcxproj" %__ExtraMsBuildParams% /nologo /maxcpucount /nodeReuse:false /p:Configuration=%__BuildType% /p:Platform=%__BuildArch% /fileloggerparameters:Verbosity=normal;LogFile="%__NativeBuildLog%"
 IF NOT ERRORLEVEL 1 goto AfterNativeBuild
 echo Native component build failed. Refer !__NativeBuildLog! for details.
 exit /b 1
 
+:BuildNativeEmscripten
+:: TODO: Add a real wasm build
```

Comment:
Can you give us an idea of what you have in mind here? I know msbuild and clang might not match up too well but I'd like to understand what we'd do instead on Windows.

Reply:
It looks like there has been some work in the past to create an extension for VS to allow building with Emscripten, however this looks to be VS 2010 only. (http://kripken.github.io/emscripten-site/docs/getting_started/getting_started_with_emscripten_and_vs2010.html). Whilst it would be nice to keep it all under msbuild on Windows, I am not sure at the moment how easy that will be.

I have however, been able to get the same build progress as on linux by installing GNU make on Windows (http://gnuwin32.sourceforge.net/packages/make.htm).

Generation
```
emcmake "%CMakePath%" "-DCMAKE_TOOLCHAIN_FILE=%EMSCRIPTEN%/cmake/Modules/Platform/Emscripten.cmake" "-DCMAKE_MAKE_PROGRAM=C:\Program Files (x86)\GnuWin32\bin\make.exe" "-DCMAKE_BUILD_TYPE=Debug" -G "MinGW Makefiles" %1
```

Build
```
pushd "%__IntermediatesDir%"
emmake "C:\Program Files (x86)\GnuWin32\bin\make.exe" install
popd
```

Reply:
I think we typically use the Windows build of CMake instead of GNU make https://cmake.org/download/ . Hopefully that still fits everything else you're doing.

Reply:
As CMake is only generating the build system, we still need to pick a generator. The normal Windows build is using the msbuild generator, but in order to use msbuild with Emscripten, more work would be required (although this would be cleaner). At the moment, I was using the Make generator, as it fits in nicely with the Emscripten build (although adds a dependency on Make).

Reply:
Ah, sorry for the confusion. We might be able to try NMake, which comes with Visual Studio and is supposedly supported by cmake on Windows. (MS folks might have licensing problems with GNU Make; something that comes with Visual Studio or uses a different license will be a lot easier for us).

Reply:
(To be clear, we'd happily take your change with the GNU Make dependency and then try to replace it as a separate change).

Reply:
NMake is a good shout. I had overlooked that. CMake has an NMake generator and it seems to work nicely. I've updated the scripts to use it.